### ISSUE TITLE
Automatic differentiation 

### ISSUE THREAD

**Issue by Evizero** [tbreloff/OnlineAI.jl#2]
A hot topic for neural networks is the use of automatic differentiation. I have seen the recent announcement of [ForwardDiff](https://github.com/JuliaDiff/ForwardDiff.jl) which seems very promising in that regard. Might be worth considering for this package. 

For example: Here is a [discussion](https://www.reddit.com/r/MachineLearning/comments/3ja454/mochajl_deep_learning_for_julia/) on the MachineLearning subreddit where that interest came up.

**Comment by tbreloff** [comment]
Yes nothing stopping auto diff as part of the framework... Should be no more than a short method or 2 to implement. Let me know if you have a specific cost function or activation that you're interested in... I might be able to help out.
